### PR TITLE
Avoid top-level `with ...;` in several packages in `pkgs/tools/`

### PR DESCRIPTION
**Pull request by philiptaron**
## Description of changes

Using `with` at a top-level scope is [an anti-pattern](https://nix.dev/guides/best-practices#with-scopes). The tracking issue is #208242. This is a pure refactor: there is no functional change contained in this PR.

I really wish to use `callPackage` style dependency injection for the two Python packages here, rather than the style I've used in this PR. I don't know how to do that refactor, however. 

I used the refactor strategy that looks for the uses of names [coming from `lib`, `pkgs`, `python3Packages`, and more](https://gist.github.com/philiptaron/04326000d5ff20cb72c2805efd09701c). 

Following advice, I also preferred to inherit names from `lib` instead of `builtins` where there was a choice. I also preferred to inherit from `lib` in one place, rather than to have a mixture of `with lib;`, `lib.thing` and `inherit (lib) thing`. **I can drop those changes if reviewers desire.**

There are no rebuilds when I run `nixpkgs-review rev HEAD`, so I've targeted master with this PR.

## Things done

- [x] Tested compilation of all packages that depend on this change using `nixpkgs-review rev HEAD`.
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/graphics/astc-encoder/default.nix [hunk 1]
```diff
@@ -5,11 +5,20 @@
 , simdExtensions ? null
 }:
 
-with rec {
```
Comment (fricklerhandwerk) [inline pkgs/tools/graphics/astc-encoder/default.nix:8]: You don't cease to find new atrocities

File: pkgs/tools/misc/bepasty/default.nix [hunk 2]
```diff
@@ -18,10 +19,41 @@ let
     };
   };
 
-#We need to use buildPythonPackage here to get the PYTHONPATH build correctly.
-#This is needed for services.bepasty
-#https://github.com/NixOS/nixpkgs/pull/38300
-in with bepastyPython.pkgs; buildPythonPackage rec {
+  inherit (lib) licenses maintainers;
+
+  inherit (bepastyPython.pkgs)
```
Comment (fricklerhandwerk) [inline pkgs/tools/misc/bepasty/default.nix:24]: Hm no, quite clearly I'd put the `with bepastyPython.pkgs;` in front of the `buildInputs` (and such) lists. If you want to be fancy, you can do some `attrValues { inherit (...) ...; }`, but that amounts to the same thing in this particular case.

**Review by fricklerhandwerk — CHANGES_REQUESTED** [review]

File: pkgs/tools/graphics/astc-encoder/default.nix (hunk 1 above)
Comment (philiptaron) [reply pkgs/tools/graphics/astc-encoder/default.nix:8]: It's a thing! It's a way to do it!

**Comment by philiptaron** [comment]
Ugh. [This](https://github.com/NixOS/nixpkgs/actions/runs/8462117682/job/23182967347?pr=299284) is why I don't like doing ad-hoc, non-tool-assisted refactors -- applying CR feedback ends up causing regressions.

**Comment by philiptaron** [comment]
@fricklerhandwerk, I will note this message from the PR headmatter:

> I really wish to use callPackage style dependency injection for the two Python packages here, rather than the style I've used in this PR. I don't know how to do that refactor, however.

If you do know how to pull that off, _that's_ the better PR.

**Comment by fricklerhandwerk** [comment]
> If you do know how to pull that off, that's the better PR.

Check how it's done here:
- [Using `callPackageWith` with on `python3.pkgs` ](https://github.com/Nix-Security-WG/nix-security-tracker/blob/5a72910c7a65a159d964c593e433ad3c10f5fa22/pkgs/python/default.nix#L4-L6)
- [Sample package recipe with Python inputs](https://github.com/Nix-Security-WG/nix-security-tracker/blob/main/pkgs/python/pyngo/default.nix)

**Review by fricklerhandwerk — APPROVED** [review]

**Comment by philiptaron** [comment]
Thanks! Those are great references. I'm going to pull https://github.com/NixOS/nixpkgs/pull/299022 into draft and keep it on the slow burner to see if I can make use of those patterns in nixpkgs.

**Pull request merged**